src/JsonSchema/Constraints/UndefinedConstraint.php

```diff
@@ -144,7 +144,8 @@ protected function validateCommonProperties(&$value, $schema = null, JsonPointer
             } elseif (isset($schema->required) && !is_array($schema->required)) {
                 // Draft 3 - Required attribute - e.g. "foo": {"type": "string", "required": true}
                 if ($schema->required && $value instanceof self) {
-                    $this->addError(ConstraintError::REQUIRED_D3(), $path);
+                    $propertyName = current($path->getPropertyPaths());
```

Comment:
Please use `end()` rather than `current()` - `current()` makes assumptions about the state of the internal pointer on the returned array, which could result in hard-to-find bugs if the pointer code is changed at a later date.